src/registry.rs

```diff
@@ -60,3 +55,38 @@ pub fn get_or_default(name: &RegKey, value: RegVal) -> RegVal {
     }
 }
 
+/// Set a value in the registry.
+///
+/// If the key already exists, returns the old value.
+/// Returns `None` for new keys.
```

Comment:
These seem like confusing return values. Why should I care what the old value is when I set it?